src/spikeinterface/sortingcomponents/benchmark/benchmark_matching.py

```diff
@@ -16,101 +17,483 @@
+        self.sort_folders.append(sort_folder)
+
+        # Generate New Waveform Extractor with Misclassed Spike Trains
+        we = extract_waveforms(self.recording, gt_sorting, self.tmp_folder, **we_kwargs)
+        methods_kwargs = self.update_methods_kwargs(we, template_mode)
+
+        sortings, _ = self.run_matching(methods_kwargs, we.unit_ids)
+        shutil.rmtree(self.tmp_folder)
+        return sortings, gt_sorting
+
+    def run_matching_missing_units(
+        self, fraction_missing, snr_threshold=0, seed=0, we_kwargs=None, template_mode="median"
+    ):
+        """Run template matching with a given fraction of missing units.
+
+        Parameters
+        ----------
+        fraction_missing: float
+            The fraction of missing units.
+        snr_threshold: float
+            The SNR threshold below which units are considered missing. (Default: 0)
+        seed: int
+            Random seed. (Default: 0)
+        we_kwargs: dict
+            A dictionary of keyword arguments for the WaveformExtractor.
+        template_mode: {'mean' | 'median' | 'std'}
+            The mode to use to extract templates from the WaveformExtractor. (Default: 'median')
+
+        Returns
+        -------
+        sortings: dict
+            A dictionary that maps method --> NumpySorting.
+        gt_sorting: NumpySorting
+            The ground-truth sorting used for template matching (with missing units).
+        """
+        try:
+            assert 0 <= fraction_missing <= 1
+        except AssertionError:
+            raise ValueError("'fraction_missing' must be between 0 and 1.")
+        try:
+            assert snr_threshold >= 0
+        except AssertionError:
+            raise ValueError("'snr_threshold' must be greater than or equal to 0.")
+        if we_kwargs is None:
+            we_kwargs = {}
+        we_kwargs.update(dict(seed=seed, overwrite=True, load_if_exists=False, **self.job_kwargs))
+        rng = np.random.default_rng(seed)
+
+        # Omit fraction_missing of units with lowest SNR
+        metrics = self.metrics.sort_values("snr")
+        missing_units = np.array(metrics.index[metrics.snr < snr_threshold])
+        num_missing = int(len(missing_units) * fraction_missing)
+        missing_units = rng.choice(missing_units, size=num_missing, replace=False)
+        present_units = np.setdiff1d(self.we.unit_ids, missing_units)
+        spike_time_indices, spike_cluster_ids = [], []
+        for unit in present_units:
+            spike_train = self.gt_sorting.get_unit_spike_train(unit)
+            for time_index in spike_train:
+                spike_time_indices.append(time_index)
+                spike_cluster_ids.append(unit)
+        spike_time_indices = np.array(spike_time_indices)
+        spike_cluster_ids = np.array(spike_cluster_ids)
+        gt_sorting = NumpySorting.from_times_labels(
+            spike_time_indices, spike_cluster_ids, self.sampling_rate, unit_ids=present_units
```

Comment:
We have a method that do the same I think.
`sorting_short = sorting.select_units(some_ids)`
can you check ?